wen-restart/src/wen_restart.rs

```diff
@@ -281,18 +301,127 @@ pub(crate) fn find_heaviest_fork(
                     );
                 }
                 return Err(WenRestartError::BlockNotLinkedToExpectedParent(
-                    slot,
+                    *slot,
                     block_meta.parent_slot,
                     expected_parent,
                 )
                 .into());
             }
-            expected_parent = slot;
+            if !block_meta.is_full() {
+                return Err(WenRestartError::BlockNotFull(*slot).into());
+            }
+            expected_parent = *slot;
         } else {
-            return Err(WenRestartError::BlockNotFound(slot).into());
+            return Err(WenRestartError::BlockNotFound(*slot).into());
+        }
+    }
+    let heaviest_fork_bankhash = find_bankhash_of_heaviest_fork(
+        heaviest_fork_slot,
+        slots,
+        blockstore.clone(),
+        bank_forks.clone(),
+        root_bank,
+        &exit,
+    )?;
+    info!(
+        "Heaviest fork found: slot: {}, bankhash: {:?}",
+        heaviest_fork_slot, heaviest_fork_bankhash
+    );
+    Ok((heaviest_fork_slot, heaviest_fork_bankhash))
+}
+
+fn find_bankhash_of_heaviest_fork(
+    heaviest_fork_slot: Slot,
+    slots: Vec<Slot>,
+    blockstore: Arc<Blockstore>,
+    bank_forks: Arc<RwLock<BankForks>>,
+    root_bank: Arc<Bank>,
+    exit: &Arc<AtomicBool>,
+) -> Result<Hash> {
+    // Find the hash of the heaviest fork, if block hasn't been replayed, replay to get the hash.
```

Comment:
This seems like it should either be a function comment or  be moved later towards where `process_single_slot()` is called

Reply:
Moved to function level.